### PR TITLE
Add grpc status, set grpc protocol

### PR DESCRIPTION
**Pull request by qiwzhang**
**What this PR does / why we need it**:

For https://github.com/istio/proxy/issues/1805

Check content type to set context.protocol correctly.
Extract grpc status and grpc message attributes from response.tailers. 

**Special notes for your reviewer**:

**Release note**:
```release-note
None
```

### CODE REVIEW

File: src/envoy/http/mixer/report_data.h [hunk 1]
```diff
@@ -84,6 +90,19 @@ class ReportData : public ::istio::control::http::ReportData {
     }
     return false;
   }
+
+  bool GetGrpcStatus(GrpcStatus *status) const override {
```
Comment (JimmyCYJ) [inline src/envoy/http/mixer/report_data.h]: IIRC, Sometimes when response body is empty, then trailers_ will be empty too. In this case grpc status and message are sent in response headers.

Comment (qiwzhang) [reply src/envoy/http/mixer/report_data.h]: Thanks.  updated the code to get grpc-status from both trailers and response headers.

File: src/envoy/http/mixer/report_data.h [hunk 2]
```diff
@@ -84,6 +104,13 @@ class ReportData : public ::istio::control::http::ReportData {
...
+    // Check trailer first.
+    // If not response body, grpc-status is in response headers.
+    return ExtractGrpcStatus(trailers_, status) ||
+           ExtractGrpcStatus(headers_, status);
```
[the remark this answers is not in the document]
Comment (qiwzhang) [reply src/envoy/http/mixer/report_data.h]: No, they are always together.  Please see: https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md

**Pull request merged**